packages/sitecore-jss/src/personalize/graphql-personalize-service.ts

```diff
@@ -46,6 +47,7 @@ type PersonalizeQueryResult = {
 
 export class GraphQLPersonalizeService {
   private graphQLClient: GraphQLClient;
+  private memoryCache: MemoryCacheClient<PersonalizeQueryResult>;
```

Comment:
We should use the `CacheClient` interface instead to allow overriding the cache implementation if desired. We can use `MemoryCacheClient` by default (i.e. in a protected `getCacheClient()` method). We should also extend the `GraphQLPersonalizeServiceConfig` with `CacheOptions` and use these when initializing (enabled by default).

Please take a look at the [DictionaryServiceBase](https://github.com/Sitecore/jss/blob/dev/packages/sitecore-jss/src/i18n/dictionary-service.ts) for an example.

Reply:
Done